packages/ibm-products/src/components/FilterPanel/FilterPanelCheckboxWithOverflow/FilterPanelCheckboxWithOverflow.js

```diff
@@ -0,0 +1,181 @@
+import { pkg } from '../../../settings';
+
+// The block part of our conventional BEM class names (blockClass__E--M).
+const blockClass = `${pkg.prefix}--filter-panel-checkbox-with-overflow`;
+const componentName = 'FilterPanelCheckboxWithOverflow';
+
+const defaults = {
+  overflowMenuProps: {},
+};
+
+/**
+ * Provides a filter panel checkbox with an overflow menu.
+ *
+ * This component's props refer to the Carbon Checkbox,
+ * just like FilterPanelCheckbox.
+ *
+ * Use `overflowMenuProps` for properties specific to the OverflowMenu.
+ */
+export let FilterPanelCheckboxWithOverflow = React.forwardRef(
+  (
+    {
+      children,
+      className,
+      count,
+      id,
+      labelText,
+      overflowMenuProps = defaults.overflowMenuProps,
+      title,
+      ...rest
+    },
+    ref
+  ) => {
+    const [menuIsOpen, setMenuIsOpen] = useState(false);
+    const [showMenuButton, setShowMenuButton] = useState(false);
+    // Destructure overflow menu properties.
+    const { ...other } = overflowMenuProps;
+
+    const backupRef = useRef();
+    const localRef = ref || backupRef;
+
+    const setHideButton = (activeElement) => {
+      if (!menuIsOpen && !localRef.current.contains(activeElement)) {
+        setShowMenuButton(false);
+      }
+    };
+
+    useEffect(() => {
+      setShowMenuButton(menuIsOpen);
+    }, [menuIsOpen]);
+
+    return (
+      // eslint-disable-next-line jsx-a11y/no-static-element-interactions
+      <div
+        className={cx(blockClass, className, {
+          [`${blockClass}--open`]: menuIsOpen,
+        })}
+        onBlur={({ relatedTarget }) => setHideButton(relatedTarget)}
+        onFocus={() => setShowMenuButton(true)}
+        /**
+         * onMouseDown > preventDefault solves an issue.
+         * Desired behavior: onMouseEnter shows the menu, even when the user mouse-presses and -releases the checkbox/label.
+         * Issue: onMouseDown > checkbox/label triggers an onBlur event and hides the menu button.
+         *   Releasing the mouse button then shows the button again.
+         *   This behavior "flickers" the menu button.
+         * Solution: onMouseDown > preventDefault prevents onBlur from being called.
+         *   The menu button remains visible at all times.
+         *   This applies only when using the mouse; it is not an issue using the keyboard.
+         */
+        onMouseDown={(event) => event.preventDefault()}
+        onMouseEnter={() => setShowMenuButton(true)}
+        onMouseLeave={() => setHideButton(document.activeElement)}
+        ref={localRef}
+        {...getDevtoolsProps(componentName)}
+      >
+        <FilterPanelCheckbox
+          {...rest}
+          className={cx(`${blockClass}__checkbox`)}
+          count={count}
+          id={id}
+          labelText={labelText}
+          title={title}
+        />
+
+        {showMenuButton && (
+          <OverflowMenu
+            open={overflowMenuProps?.open ? true : false}
```

Comment:
This should also work

```suggestion
            open={overflowMenuProps?.open}
```